opentelemetry-jaeger/src/lib.rs

```diff
@@ -484,11 +484,18 @@ mod propagator {
 
         /// Extract span id from the header.
         fn extract_span_id(&self, span_id: &str) -> Result<SpanId, ()> {
-            if span_id.len() != 16 {
-                return Err(());
+            match span_id.len() {
+                // exact 16
+                16 => SpanId::from_hex(span_id).map_err(|_| ()),
+                // more than 16 is invalid
+                17.. => Err(()),
+                // less than 16 will result padding on left
+                _ => {
+                    let mut padded = String::with_capacity(16 - span_id.len());
```

Comment:
`with_capacity` will allocate enough room but not left pad. I think format may be easier? 

```rust
let padded = format!("{span_id:0>16}");
```

Reply:
Good point. Let me revise it